### PR TITLE
eventPage

### LINKED ISSUE #27: Event oldal elkészítése
Idk hogy ez az issue miért nem létezett eddig, pedig ezen dolgoztam eddig is xd

### PR DESCRIPTION
**Pull request by SzBeni2003**
A working page for showing details about an event, using data from our database. Could be cleaner, but it works

### CODE REVIEW

File: apps/frontend/src/api/hooks/eventMutationHooks.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
+
+import { EventModel } from '@/api/model/event.model';
+import { CreateEvent } from '@/app/events/types/createEvent';
+
+const url = 'http://localhost:3001';
```
Comment (DannySS123) [inline apps/frontend/src/api/hooks/eventMutationHooks.ts]: this could be an env var

File: apps/frontend/src/api/hooks/eventQueryHooks.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use client';
+import { useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+
+import { EventDetails } from '@/app/events/types/eventDetails';
+
+const url = 'http://localhost:3001';
```
Comment (DannySS123) [inline apps/frontend/src/api/hooks/eventQueryHooks.ts]: And could be used here too

File: apps/frontend/src/api/hooks/eventQueryHooks.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+export const useFetchEventDetailsQuery = (eventId: string) => {
+  return useQuery<EventDetails>({
+    queryKey: ['fetchEventDetails', eventId],
+    queryFn: async () => (await axios.get<EventDetails>(`http://localhost:3001/events/${eventId}`)).data,
```
Comment (DannySS123) [inline apps/frontend/src/api/hooks/eventQueryHooks.ts]: here too

File: apps/frontend/src/app/events/new/page.tsx [hunk 4]
```diff
@@ -1,8 +1,56 @@
...
+
+import { useCreateEventMutation } from '@/api/hooks/eventMutationHooks';
+import { Color, EventModel, Status } from '@/api/model/event.model';
+import { CreateEvent, CreateEventForm } from '@/app/events/types/createEvent';
 import Button from '@/components/button';
 import Input from '@/components/input';
 import { styles } from '@/components/newEventStyles';
 
 export default function newEvent() {
+  /*const { mutate: createEvent } = useCreateEventMutation();
```
Comment (DannySS123) [inline apps/frontend/src/app/events/new/page.tsx:12]: What about this comment?

Comment (SzBeni2003) [reply apps/frontend/src/app/events/new/page.tsx:12]: For the testing I wanted to connect the event creation page to the api, but was unsuccessful. That will be my next priority on a different branch, to be able to actually create events from the event creation page.

File: apps/backend/src/main.ts [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
+import cors from 'cors';
 
 import { AppModule } from './app.module';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
   app.useGlobalPipes(new ValidationPipe({ whitelist: true }));
+  app.use(
+    cors({
+      origin: 'http://localhost:3000',
```
Comment (DannySS123) [inline apps/backend/src/main.ts]: lets use an env var here as well

**Review by DannySS123 — APPROVED** [review]
👌

**Pull request merged**